deployments/helm/gpud/templates/daemonset.yaml

```diff
@@ -131,23 +286,106 @@ spec:
             {{- toYaml .Values.resources | nindent 12 }}
 
           volumeMounts:
+            # System logs - GPUd writes logs here and monitors system logs
             - name: host-log
               mountPath: /var/log
+
+            # Kernel crash dumps and persistent storage for error analysis
             - name: host-pstore
               mountPath: /var/lib/systemd/pstore
+
+            # Device files - Critical for:
+            # - /dev/kmsg: kernel message buffer (pkg/kmsg/watcher.go)
+            # - /dev/nvidia*: GPU devices for NVIDIA monitoring
+            # - /dev/dri/*: GPU DRM devices
             - name: host-dev
               mountPath: /dev
+
+            # Sysfs - Hardware/device info (read-only for safety)
+            # - /sys/class/infiniband: InfiniBand network monitoring
+            # - /sys/class/dmi: Hardware/DMI information
+            # - /sys/fs/fuse/connections: FUSE filesystem stats
+            - name: host-sys
+              mountPath: /sys
+              readOnly: true
+
+            # Procfs - Kernel/process info (read-only for safety)
+            # - /proc/sys/fs/file-*: File descriptor limits and usage
+            # - /proc/sys/kernel/random/boot_id: Unique boot identifier
+            # - /proc/self/mountinfo: Current mount points
+            - name: host-proc
+              mountPath: /proc
+              readOnly: true
```

Comment:
Mounting both of these directories come with some unfortunate security considerations. But I'm not sure there is a way around this if the user wants to track these metrics.

It could make sense to wrap these in optional values as well, perhaps default `true`, to give the user control to opt-out of mounting these host directories in sensitive environments if they so choose.

That is, so long as `gpud` continues to function as you expect without access to these directories.